cppwinrt/text_writer.h

```diff
@@ -157,9 +157,16 @@ namespace cppwinrt
         {
             if (!file_equal(filename))
             {
-                std::ofstream file{ filename, std::ios::out | std::ios::binary };
-                file.write(m_first.data(), m_first.size());
-                file.write(m_second.data(), m_second.size());
+                std::ofstream file;
+                file.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+                try {
+                  file.open(filename, std::ios::out | std::ios::binary);
+                  file.write(m_first.data(), m_first.size());
+                  file.write(m_second.data(), m_second.size());
+                }
+                catch (std::ofstream::failure const& e) {
+                  throw std::filesystem::filesystem_error(e.what(), filename, std::io_errc::stream);
```

Comment:
Why rethrow?

Reply:
the `ofstream::failure` does not have context about what file failed, so the idea is to throw a filesystem_error that does have the ability to point to what file is the problem, and main will just print the message that contains the file path